src/VCR/Videorecorder.php

```diff
@@ -216,7 +216,10 @@ public function handleRequest(Request $request)
             );
         }
 
+        $this->getMatchObserver()->startObserving();
+
         if ($this->cassette->hasResponse($request)) {
+            $this->getMatchObserver()->stopObserving();
```

Comment:
I agree. I didn't want to "mess too much" with the existing code/order, but I think that was my mistake as a novice open-source contributor. What I ended up doing made less sense. I will use your idea.